packages/nx/src/plugins/js/utils/packages.ts

```diff
@@ -15,13 +15,28 @@ export function getPackageEntryPointsToProjectMap<
     }
 
     const { packageName, packageExports } = metadata.js;
-    if (!packageExports || typeof packageExports === 'string') {
+    if (
+      !packageExports ||
+      typeof packageExports === 'string' ||
+      !Object.keys(packageExports).length
+    ) {
       // no `exports` or it points to a file, which would be the equivalent of
       // an '.' export, in which case the package name is the entry point
       result[packageName] = project;
     } else {
       for (const entryPoint of Object.keys(packageExports)) {
-        result[join(packageName, entryPoint)] = project;
+        // if entrypoint begins with '.', it is a relative subpath export
+        // otherwise, it is a conditional export
+        // https://nodejs.org/api/packages.html#conditional-exports
+        if (entryPoint.startsWith('.')) {
+          result[join(packageName, entryPoint)] = project;
+        } else {
+          result[packageName] = project;
+        }
+      }
+      // if there was no '.' entrypoint, ensure the package name is matched with the project
+      if (!result[packageName]) {
+        result[packageName] = project;
```

Comment:
I considered an approach like this, but I wasn't sure if this map was used elsewhere and this could be misleading.

Reply:
Yeah, this is only used to identify dependencies when building the project graph, so it doesn't need to be that strict. If the user has a wrong import path that is not allowed by the definition of exports, the build tool will complain, and they'll have to fix it anyway.